cpp/src/fil/infer.cu

```diff
@@ -309,17 +318,17 @@ struct tree_aggregator_t<NITEMS, CATEGORICAL_LEAF> {
   int num_classes;
 
   static size_t smem_finalize_footprint(int num_classes) {
-    return sizeof(int) * num_classes * NITEMS;
+    // not accounting for lingering accumulate_footprint during finalize()
+    return 0;
   }
   static size_t smem_accumulate_footprint(int num_classes) {
-    return smem_finalize_footprint(num_classes);
```

Comment:
also misleading, as the accumulate_workspace variables have a different offset than finalize_workspace variables.